docs/docusaurus/scripts/remark-named-snippets/snippet.js

```diff
@@ -191,27 +189,33 @@ function processVerbose() {
 }
 
 function main(verbose = false) {
-  const snippets = parseSourceDirectories(getDirs())
   let argNum = 2
   if (verbose) { argNum = 3 }
   const targetFiles = process.argv.slice(argNum)
 
-  const out = {}
-  for (const snippet of snippets) {
-    // If no explicit args are provided, default to all snippets
-    // Else, ensure that the snippet's source file was requested by the user
-    const file = snippet.file
-    if (targetFiles.length > 0 && !(targetFiles.includes(file))) {
-      continue
+  const dirs = getDirs()
+  for (let dir of dirs) {
+    const snippets = parseSourceDirectories(dir)
+    const out = {}
+    for (const snippet of snippets) {
+      // If no explicit args are provided, default to all snippets
+      // Else, ensure that the snippet's source file was requested by the user
+      const file = snippet.file
+      if (targetFiles.length > 0 && !(targetFiles.includes(file))) {
+        continue
+      }
+      if (!(file in out)) {
+        out[file] = []
+      }
+      delete snippet.file // Remove duplicate filename to clean up stdout
+      out[file].push(snippet)
     }
-    if (!(file in out)) {
-      out[file] = []
+    if (verbose) {
+      // printing objects one dir at a time so the output is more readable
+      console.log("Snippets found in " + dir)
+      console.log("---------------------------")
+      console.log(out)
```

Comment:
Will this print legibly to the console or do we need to iterate through?

Reply:
It prints legibly. I tried putting `out` for each directory into a larger dictionary and started seeing `[Object object]` (or something like that) for leaf nodes.